main.go

```diff
@@ -0,0 +1,129 @@
+
+		// The default controller mode.
+		controllerCmd = app.Command(filepath.Base(os.Args[0]), "An open source multicloud control plane.").Default()
+
+		// Configuration for the reconciler.
+		reconcileCmd                = controllerCmd.Command("reconcile", "Reconcile a Custom Resource")
+		templateStackNameInput      = controllerCmd.Flag("template-stack-name", "Name of the TemplateStack custom resource.").Required().String()
+		templateStackNamespaceInput = controllerCmd.Flag("template-stack-namespace", "Namespace of the TemplateStack custom resource").String()
+	)
+	cmd := kingpin.MustParse(app.Parse(os.Args[1:]))
+
+	kingpin.FatalIfError(clientgoscheme.AddToScheme(scheme), "could not register client-go scheme")
+	kingpin.FatalIfError(stacksv1alpha1.AddToScheme(scheme), "could not register stacks group scheme")
+
+	switch cmd {
+	case reconcileCmd.FullCommand():
+		mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+			Scheme: scheme,
+			Port:   9443,
+		})
+		kingpin.FatalIfError(err, "unable to start manager")
+
+		ts := &stacksv1alpha1.TemplateStack{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      *templateStackNameInput,
+				Namespace: *templateStackNamespaceInput,
+			},
+		}
+		kingpin.FatalIfError(getTemplateStack(ts), "could not fetch the TemplateStack object")
+
+		var options []controllers.ResourcePackReconcilerOption
+		switch ts.Spec.Behavior.EngineConfiguration.Type {
+		case KustomizeEngine:
+
+			// TODO(muvaf): investigate a better way to convert *Unstructured to *Kustomization.
+			kustomizationYAML, err := yaml.Marshal(ts.Spec.Behavior.EngineConfiguration.Kustomization)
+			kingpin.FatalIfError(err, "cannot marshal kustomization object")
+			kustomization := &kustomizeapi.Kustomization{}
+			kingpin.FatalIfError(yaml.Unmarshal(kustomizationYAML, kustomization), "cannot unmarshal into kustomization object")
+
+			options = append(options, controllers.WithTemplatingEngine(kustomize.NewKustomizeEngine(kustomization,
+				kustomize.WithResourcePath(ts.Spec.Behavior.Source.Path),
+				kustomize.AdditionalOverlayGenerator(kustomize.NewPatchOverlayGenerator(ts.Spec.Behavior.EngineConfiguration.Overlays)),
+			)))
+		}
+		gvk := schema.FromAPIVersionAndKind(ts.Spec.Behavior.CRD.APIVersion, ts.Spec.Behavior.CRD.Kind)
+		controller := controllers.NewResourcePackReconciler(mgr, gvk, options...)
+		u := &unstructured.Unstructured{}
+		u.SetGroupVersionKind(gvk)
+		kingpin.FatalIfError(
+			ctrl.NewControllerManagedBy(mgr).
+				For(u).
+				Complete(controller),
+			"could not create controller",
+		)
+		kingpin.FatalIfError(clientgoscheme.AddToScheme(scheme), "could not register client go scheme")
+		kingpin.FatalIfError(stacksv1alpha1.AddToScheme(scheme), "could not register template stack scheme")
+		kingpin.FatalIfError(mgr.Start(ctrl.SetupSignalHandler()), "unable to run the manager")
+	}
+}
+
+// TODO: Controller-runtime client doesn't work until manager is started, which
```

Comment:
What if we injected it using a config map volume mount then? Shrug. Just an idea

Reply:
Yeah I considered that, too. But it'd increase the number of resources if we still keep a `TemplateStack`-like object around. I think when we move to a more unified experience like only two kinds existing `TemplateStack and Stack` or only one, we can consider creating a `ConfigMap` with necessary info and mount it.